ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/JarManifestErrorReporter.java

```diff
@@ -193,6 +206,25 @@ protected void parseManifest(IDocument document, IProgressMonitor monitor) {
 				}
 
 			}
+			if (emptyLines.size() > 0) {
+				// build a marker with all the line info aggregated into it,
+				// with the first empty line as the anchor line.
+				// could have been easy if the marker accepts maps, but it
+				// doesn't at the moment. TODO: enhance marker to process maps.
+				StringBuilder str = new StringBuilder();
```

Comment:
You could use StringJoiner like
```
StringJoiner str = new StringJoiner(","); //$NON-NLS-1$
for (Integer i : emptyLines) {
	str.add(Integer.toString(i));
}
```
and so don't need to add separators / remove last one

Reply:
thanks, I used this class to compose the marker string.